crates/modelardb_server/src/configuration.rs

```diff
@@ -0,0 +1,154 @@
+            .await;
+
+        self.uncompressed_reserved_memory_in_bytes = new_uncompressed_reserved_memory_in_bytes;
+    }
+
+    pub(crate) fn compressed_reserved_memory_in_bytes(&self) -> &usize {
+        &self.compressed_reserved_memory_in_bytes
+    }
+
+    /// Set the new value and update the compressed remaining reserved memory in the storage engine.
+    /// If the value was updated, return [`Ok`], otherwise return
+    /// [`ConfigurationError`](ModelarDbError::ConfigurationError).
+    pub(crate) async fn set_compressed_reserved_memory_in_bytes(
+        &mut self,
+        new_compressed_reserved_memory_in_bytes: usize,
+        storage_engine: Arc<RwLock<StorageEngine>>,
+    ) -> Result<(), ModelarDbError> {
+        let value_change = new_compressed_reserved_memory_in_bytes as isize
+            - self.compressed_reserved_memory_in_bytes as isize;
+
+        storage_engine
+            .write()
+            .await
+            .set_compressed_remaining_memory_in_bytes(value_change)
+            .await
+            .map_err(|error| ModelarDbError::ConfigurationError(error.to_string()))?;
+
+        self.compressed_reserved_memory_in_bytes = new_compressed_reserved_memory_in_bytes;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::common_test;
+
+    // Tests for ConfigurationManager.
+    #[tokio::test]
+    async fn test_set_uncompressed_reserved_memory_in_bytes() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let context = common_test::test_context(temp_dir.path()).await;
+
+        let configuration_manager = context.configuration_manager.clone();
+        let storage_engine = context.storage_engine.clone();
+
+        assert_eq!(
+            *configuration_manager.read().await.uncompressed_reserved_memory_in_bytes(),
+            common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES
+        );
+
+        configuration_manager
+            .write()
+            .await
+            .set_uncompressed_reserved_memory_in_bytes(1024, storage_engine)
+            .await;
+
+        assert_eq!(
+            *configuration_manager.read().await.uncompressed_reserved_memory_in_bytes(),
+            1024
+        );
+    }
+
+    #[tokio::test]
+    async fn test_set_compressed_reserved_memory_in_bytes() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let context = common_test::test_context(temp_dir.path()).await;
+
+        let configuration_manager = context.configuration_manager.clone();
+        let storage_engine = context.storage_engine.clone();
+
+        assert_eq!(
+            *configuration_manager.read().await.compressed_reserved_memory_in_bytes(),
+            common_test::COMPRESSED_RESERVED_MEMORY_IN_BYTES
+        );
+
+        configuration_manager
+            .write()
+            .await
+            .set_compressed_reserved_memory_in_bytes(1024, storage_engine)
+            .await
+            .unwrap();
```

Comment:
Since an error is only returned if compressed data needed to be saved to disk and an error happened either when saving to Parquet or when adding the file to the data transfer component, I do not think it makes sense to add another test for this case here.